File: apps/backend/app/schemas/purchase_metrics.py
# backend/app/schemas/purchase_metrics.py
from pydantic import BaseModel, ConfigDict
from datetime import date
from typing import List

class SaleByDay(BaseModel):
    day: date
    num_purchases: int
    revenue: float
    model_config = ConfigDict(from_attributes=True)


class PurchaseMetricRead(BaseModel):
    start_date: date
    end_date: date
    total_purchases: int
    total_sales: float
    avg_ticket: float
    unique_buyers: int
    avg_purchases_per_user: float
    sales_by_day: List[SaleByDay]
    model_config = ConfigDict(from_attributes=True)

class PurchasesPerUser(BaseModel):
    user_id: str
    purchase_count: int
    model_config = ConfigDict(from_attributes=True)

class RevenuePerUser(BaseModel):
    user_id: str
    total_spent: float
    model_config = ConfigDict(from_attributes=True)

File: apps/portal-company/src/types/purchaseMetrics.ts
// src/types/purchaseMetrics.ts

/** Série diária de compras (nº e receita) */
export interface SaleByDay {
  day: string;        // ex: "2025-07-09"
  num_purchases: number;
  revenue: number;
}

/** Métricas de compras */
export interface PurchaseMetricRead {
  start_date: string;             // "YYYY-MM-DD"
  end_date: string;               // "YYYY-MM-DD"
  total_purchases: number;
  total_sales: number;
  avg_ticket: number;
  unique_buyers: number;
  avg_purchases_per_user: number;
  sales_by_day: SaleByDay[];
}

/** Total de compras por usuário */
export interface PurchasesPerUser {
  user_id: string;
  purchase_count: number;
}

/** Receita por usuário */
export interface RevenuePerUser {
  user_id: string;
  total_spent: number;
}